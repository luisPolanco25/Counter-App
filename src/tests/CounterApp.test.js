import React from 'react';
import {shallow} from 'enzyme';
import CounterApp from '../CounterApp';

describe('Tests in Counter App', () => {

    let wrapper = shallow(<CounterApp />);
    
    beforeEach(() => {
        wrapper = shallow(<CounterApp />);
    });

    test('CounterApp should match with its snapshot', () => {
        
        expect(wrapper).toMatchSnapshot();

    });
    
    test("CounterApp's text should match with the number provided", () => {
        
        const wrapper = shallow(<CounterApp value={100}/>);
        const value = wrapper.find('h2').text();
        
        expect(value).toBe('100');

    })

    test('should increase by touching the button +1', () => {

        wrapper.find('button').at(0).simulate('click');

        const counterText = wrapper.find('h2').text().trim();
        
        expect(counterText).toBe('1');

    });

    test('should decrease by touching the button -1', () => {

        wrapper.find('button').at(2).simulate('click');

        const counterText = wrapper.find('h2').text().trim();
        
        expect(counterText).toBe('-1');
    });
    
    test('should reset by touching the button reset', () => {

        const wrapper = shallow(<CounterApp value={105}/>);
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text().trim();

        expect(counterText).toBe('105');
    });

})
