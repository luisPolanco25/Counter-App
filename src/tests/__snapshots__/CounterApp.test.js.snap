// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Counter App CounterApp should match with its snapshot 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
